system/jlib/jlz4.cpp

```diff
@@ -0,0 +1,655 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "platform.h"
+#include "jlz4.hpp"
+#include "../lz4/lz4.h"
+
+#define COMMITTED ((size32_t)-1)
+
+/* Format:
+    size32_t totalexpsize;
+    { size32_t subcmpsize; bytes subcmpdata; }
+    size32_t trailsize; bytes traildata;    // unexpanded
+*/
+
+class jlib_decl CLZ4Compressor : public CInterface, public ICompressor
```

Comment:
Slightly better to use CSimpleInterfaceOf<ICompressor> and dispense of the IMPLEMENT_IINTERFACE;